diff -up gnutls-3.8.1/config.h.in.2~ gnutls-3.8.1/config.h.in
--- gnutls-3.8.1/config.h.in.2~	2023-08-05 00:13:18.350065392 +0200
+++ gnutls-3.8.1/config.h.in	2023-08-05 00:14:00.306460767 +0200
@@ -2362,7 +2362,7 @@
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/gl/stdio.in.h.2~ gnutls-3.8.1/gl/stdio.in.h
--- gnutls-3.8.1/gl/stdio.in.h.2~	2023-08-04 10:07:49.000000000 +0200
+++ gnutls-3.8.1/gl/stdio.in.h	2023-08-05 00:13:18.350065392 +0200
@@ -95,7 +95,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/gl/string.in.h.2~ gnutls-3.8.1/gl/string.in.h
--- gnutls-3.8.1/gl/string.in.h.2~	2023-08-04 10:07:49.000000000 +0200
+++ gnutls-3.8.1/gl/string.in.h	2023-08-05 00:13:18.350065392 +0200
@@ -72,7 +72,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/gl/wchar.in.h.2~ gnutls-3.8.1/gl/wchar.in.h
--- gnutls-3.8.1/gl/wchar.in.h.2~	2023-08-04 10:07:49.000000000 +0200
+++ gnutls-3.8.1/gl/wchar.in.h	2023-08-05 00:13:18.350065392 +0200
@@ -88,7 +88,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/m4/gnulib-common.m4.2~ gnutls-3.8.1/m4/gnulib-common.m4
--- gnutls-3.8.1/m4/gnulib-common.m4.2~	2023-08-05 00:13:18.350065392 +0200
+++ gnutls-3.8.1/m4/gnulib-common.m4	2023-08-05 00:14:20.886654880 +0200
@@ -194,7 +194,7 @@ AC_DEFUN([gl_COMMON_BODY], [
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/src/gl/string.in.h.2~ gnutls-3.8.1/src/gl/string.in.h
--- gnutls-3.8.1/src/gl/string.in.h.2~	2023-08-04 10:08:39.000000000 +0200
+++ gnutls-3.8.1/src/gl/string.in.h	2023-08-05 00:13:18.350065392 +0200
@@ -72,7 +72,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/src/gl/wchar.in.h.2~ gnutls-3.8.1/src/gl/wchar.in.h
--- gnutls-3.8.1/src/gl/wchar.in.h.2~	2023-08-04 10:08:40.000000000 +0200
+++ gnutls-3.8.1/src/gl/wchar.in.h	2023-08-05 00:13:18.350065392 +0200
@@ -88,7 +88,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up gnutls-3.8.1/src/gl/c++defs.h.omv~ gnutls-3.8.1/src/gl/c++defs.h
--- gnutls-3.8.1/src/gl/c++defs.h.omv~	2023-08-05 00:29:52.107758195 +0200
+++ gnutls-3.8.1/src/gl/c++defs.h	2023-08-05 00:30:37.452211686 +0200
@@ -279,7 +279,7 @@
    causes a warning to be emitted when ::func is used but not when
    GNULIB_NAMESPACE::func is used.  func must be defined without overloaded
    variants.  */
-#if defined __cplusplus && defined GNULIB_NAMESPACE
+#if defined __cplusplus && defined GNULIB_NAMESPACE && !defined(__clang__)
 # define _GL_CXXALIASWARN(func) \
    _GL_CXXALIASWARN_1 (func, GNULIB_NAMESPACE)
 # define _GL_CXXALIASWARN_1(func,namespace) \
@@ -306,7 +306,7 @@
 /* _GL_CXXALIASWARN1 (func, rettype, parameters_and_attributes);
    causes a warning to be emitted when the given overloaded variant of ::func
    is used but not when GNULIB_NAMESPACE::func is used.  */
-#if defined __cplusplus && defined GNULIB_NAMESPACE
+#if defined __cplusplus && defined GNULIB_NAMESPACE && !defined(__clang__)
 # define _GL_CXXALIASWARN1(func,rettype,parameters_and_attributes) \
    _GL_CXXALIASWARN1_1 (func, rettype, parameters_and_attributes, \
                         GNULIB_NAMESPACE)
