diff -Naur gnutls-3.6.3/configure gnutls-3.6.3.tpg/configure
--- gnutls-3.6.3/configure	2018-07-15 16:36:31.000000000 +0000
+++ gnutls-3.6.3.tpg/configure	2018-07-20 09:26:47.081018362 +0000
@@ -54189,7 +54189,7 @@
 shlibpath_overrides_runpath=unknown
 version_type=none
 dynamic_linker="$host_os ld.so"
-sys_lib_dlsearch_path_spec="/lib /usr/lib"
+sys_lib_dlsearch_path_spec="/lib /usr/lib /lib64 /usr/lib64"
 need_lib_prefix=unknown
 hardcode_into_libs=no
 
