diff -Naur gnutls-3.6.8/tests/Makefile.am gnutls-3.6.8.tpg/tests/Makefile.am
--- gnutls-3.6.8/tests/Makefile.am	2019-05-23 19:04:54.000000000 +0000
+++ gnutls-3.6.8.tpg/tests/Makefile.am	2019-07-31 12:43:44.102677116 +0000
@@ -455,9 +455,9 @@
 if !WINDOWS
 ctests += tls13/post-handshake-with-cert-pkcs11 pkcs11/tls-neg-pkcs11-no-key
 tls13_post_handshake_with_cert_pkcs11_DEPENDENCIES = libpkcs11mock2.la libutils.la
-tls13_post_handshake_with_cert_pkcs11_LDADD = $(LDADD) $(LIBDL)
+tls13_post_handshake_with_cert_pkcs11_LDADD = $(LDADD) $(LIBDL) -ldl
 pkcs11_tls_neg_pkcs11_no_key_DEPENDENCIES = libpkcs11mock2.la libutils.la
-pkcs11_tls_neg_pkcs11_no_key_LDADD = $(LDADD) $(LIBDL)
+pkcs11_tls_neg_pkcs11_no_key_LDADD = $(LDADD) $(LIBDL) -ldl
 endif
 endif
 
