diff --git a/gl/intprops.h b/gl/intprops.h
index 1a44ae5..212c3d3 100644
--- a/gl/intprops.h
+++ b/gl/intprops.h
@@ -228,7 +228,7 @@
 #endif
 
 /* True if __builtin_add_overflow_p (A, B, C) works.  */
-#define _GL_HAS_BUILTIN_OVERFLOW_P (7 <= __GNUC__)
+#define _GL_HAS_BUILTIN_OVERFLOW_P (7 <= __GNUC__ && !defined(__clang__))
 
 /* The _GL*_OVERFLOW macros have the same restrictions as the
    *_RANGE_OVERFLOW macros, except that they do not assume that operands
diff --git a/lib/gnutls_int.h b/lib/gnutls_int.h
index 179d71b..9b2573f 100644
--- a/lib/gnutls_int.h
+++ b/lib/gnutls_int.h
@@ -90,7 +90,7 @@ typedef int ssize_t;
 # define attr_warn_unused_result
 #endif
 
-#if _GNUTLS_GCC_VERSION >= 70100
+#if _GNUTLS_GCC_VERSION >= 70100 && !defined(__clang__)
 # define FALLTHROUGH __attribute__ ((fallthrough))
 #else
 # define FALLTHROUGH
diff --git a/src/gl/intprops.h b/src/gl/intprops.h
index 1a44ae5..212c3d3 100644
--- a/src/gl/intprops.h
+++ b/src/gl/intprops.h
@@ -228,7 +228,7 @@
 #endif
 
 /* True if __builtin_add_overflow_p (A, B, C) works.  */
-#define _GL_HAS_BUILTIN_OVERFLOW_P (7 <= __GNUC__)
+#define _GL_HAS_BUILTIN_OVERFLOW_P (7 <= __GNUC__ && !defined(__clang__))
 
 /* The _GL*_OVERFLOW macros have the same restrictions as the
    *_RANGE_OVERFLOW macros, except that they do not assume that operands
diff --git a/src/gl/xalloc-oversized.h b/src/gl/xalloc-oversized.h
index e3068c8..08b7e1f 100644
--- a/src/gl/xalloc-oversized.h
+++ b/src/gl/xalloc-oversized.h
@@ -41,7 +41,7 @@ typedef size_t __xalloc_count_type;
    positive and N must be nonnegative.  This is a macro, not a
    function, so that it works correctly even when SIZE_MAX < N.  */
 
-#if 7 <= __GNUC__
+#if 7 <= __GNUC__ && !defined(__clang__)
 # define xalloc_oversized(n, s) \
    __builtin_mul_overflow_p (n, s, (__xalloc_count_type) 1)
 #elif 5 <= __GNUC__ && !defined __ICC && !__STRICT_ANSI__
